src/shared/telemetry/activation.ts

```diff
@@ -60,54 +66,57 @@ export async function activate(activateArguments: {
     )
 }
 
-function applyTelemetryEnabledState(telemetry: TelemetryService, toolkitSettings: SettingsConfiguration) {
-    const optInSetting = toolkitSettings.readSetting<string>(AWS_TELEMETRY_KEY) ?? SETTINGS_TELEMETRY_VALUE_USEIDE
-
-    if (optInSetting === SETTINGS_TELEMETRY_VALUE_ENABLE) {
-        telemetry.telemetryEnabled = true
-    } else if (optInSetting === SETTINGS_TELEMETRY_VALUE_DISABLE) {
-        telemetry.telemetryEnabled = false
-    } else {
-        telemetry.telemetryEnabled = isVsCodeTelemetryEnabled()
+export function isTelemetryEnabled(toolkitSettings: SettingsConfiguration): boolean {
+    // Setting used to be an enum, but is now a boolean.
+    // We don't have api-based strong type support, so we have to process this value manually.
+    const value = toolkitSettings.readSetting<any>(AWS_TELEMETRY_KEY)
+
+    // Handle original opt-out value (setting used to be a tri-state string value)
+    if (value === LEGACY_SETTINGS_TELEMETRY_VALUE_DISABLE) {
+        return false
+    }
+
+    // Current value is expected to be a boolean
+    if (typeof value === 'boolean') {
+        return value
     }
+
+    // Treat anything else (unexpected values, datatypes, or undefined) as opt-in
+    return true
```

Comment:
Can we condense all of this to `return !!value`?

Reply:
Favor simple, boring code over "clever" syntax to reduce the risk of error, and to increase readability.

To your example, the number 0 is something that fails the intended logic, since it would return false instead of true.